contents/javascript/VueJs/Intro.md

```diff
@@ -0,0 +1,202 @@
+* CDN:
+  - include the following into the html file:
+    `<script src="https://cdn.jsdelivr.net/npm/vue@2.5.13/dist/vue.js"</script>`
+* NPM:
+```npm
+    $ npm install vue
+```
+* CLI:
+```cli
+    $ npm install --global vue-cli  # install vue-cli
+    $ vue init webpack my-project  # create a new project using the "webpack" template
+    $ cd my-project  # install dependencies and go!
+    $ npm install
+    $ npm run dev
+```
+
+Apart from installing VueJs, you can install [VueJs development tools](https://github.com/vuejs/vue-devtools#vue-devtools) in your browser. This will allow you to inspect and debug the Vue components in your projects.
+
+## Creating a Simple Project in VueJs:
+
+Steps to print the VueJs version of "Hello World":
+
+* Import VueJs CDN and the JavaScript file in the main HTML file.
+```HTML
+  <script src="https://unpkg.com/vue@2.5.13/dist/vue.js"></script>
+  <script src="the_path_to_the_javacript_file.js"></script>
+```
+
+* Create a Vue instance in the JavaScript file; bind the instance to one of the component in our html file (the `root` element). We can only use the Vue data inside the component with id `root` as we bind it to the Vue element.
+  - In this case, only this component is working in VueJs while the rest are unaffected. This is how we progressively plug in VueJs into our projects without a complete one at a go.
+```js
+  new Vue ({
+  	el: '#root',
+  });
+```
+```HTML
+  <div id="root"></div>
+```
+
+* Specify our data(message: "Hello World") in the Vue instance.
+```js
+  data: {
+    message: "Hello World"
+  }
+```
+
+* Pass the message to the HTML file using double curly brackets.
+```HTML
+  <div id="root">
+    <h1>{{ message }}</h1>
+  </div>
+```
+
+* Open the brower and we will see "Hello World" being displayed.
+
+Putting all parts together:
+
+```HTML
+<!--This is the main HTML file-->
+<body>
+  <div id="root">
+		<h1>{{ message }}</h1>
+	</div>
+	<script src="https://unpkg.com/vue@2.5.13/dist/vue.js"></script>
+	<script src="the_path_to_the_javacript_file.js"></script>
+</body>
+```
+
+```js
+// This is inside the javacript file
+new Vue ({
+	el: '#root',
+
+  data: {
+    message: "Hello World"
+  }
+});
+```
+
+### Common VueJs syntax
+
+#### [Text](https://vuejs.org/v2/guide/syntax.html#Text)
+
+The most basic data binding is text interpolation using double curly brackets:
+```HTML
+<!--This is the main HTML file-->
+<h1>{{ message }}</h1>
+```
+
+```js
+// This is inside the javacript file
+new Vue ({
+	el: '#root',
+
+  data: {
+    message: "Hello World"
+  }
+});
+```
+The double curly brackets with contents inside will be replaced with the value of the `message` property on the corresponding data object. It will also be updated whenever the `data` object’s `message` property changes. The double curly brackets as the identifier inside it will be replaced with the value of the `message` property on the corresponding `data` object.
```

Comment:
Does the last sentence say something different from "The double curly brackets as the identifier inside it will be replaced with the value of the `message` property on the corresponding `data` object." ?